vmf/scripts/mods/vmf/modules/ui/options/vmf_options_view.lua

```diff
@@ -4052,10 +4611,11 @@ VMFOptionsView.update_search_bar = function (self)
 
     local new_search_text = KeystrokeHelper.parse_strokes(old_search_text, text_index, "insert", keystrokes)
 
-    new_search_text = string.gsub(new_search_text, "%%", "")
+    -- Prepend % [ ] ( ) . + - * ? ^ $ with % to allow users to search on special pattern characters
+    local new_search_text_escaped = new_search_text:gsub("([%%%[%]%(%)%.%+%-%*%?%^%$])", "%%%1")
```

Comment:
The pattern is a bit hard to read.

Fortunately the Lua docs say:
> Any punctuation character (even the non magic) can be preceded by a '%' when used to represent itself in a pattern. 
http://www.lua.org/manual/5.1/manual.html#5.4.1

With that in mind, here's a suggestion that also adds pseudo-fuzzy matching on spaces (caveat emptor: untested):

```suggestion
    local new_search_text_escaped = new_search_text:gsub("%p", "%%%1"):gsub("%s+", ".-")
```

---

Also worth noting that pattern matching can be disabled via an argument:

> `string.find (s, pattern [, init [, plain]])`
> [...]
> A value of true as a fourth, optional argument plain turns off the pattern matching facilities, so the function does a plain "find substring" operation, with no characters in pattern being considered "magic".
www.lua.org/manual/5.1/manual.html#pdf-string.find